kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/AssetMaintainableImpl.java

```diff
@@ -0,0 +1,111 @@
+package edu.arizona.kfs.module.cam.document;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.kns.document.MaintenanceDocument;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.service.KRADServiceLocator;
+import org.kuali.rice.krad.service.NoteService;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.sys.KFSConstants;
+
+@SuppressWarnings("deprecation")
+public class AssetMaintainableImpl extends org.kuali.kfs.module.cam.document.AssetMaintainableImpl {
+
+    private static final long serialVersionUID = 5710219294064590806L;
+
+    @Override
+    public void processAfterEdit(MaintenanceDocument document, Map<String, String[]> parameters) {
+        List<Note> notes = new ArrayList<Note>();
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.CHANGE);
+        document.setNotes(notes);
+
+        super.processAfterEdit(document, parameters);
+    }
+
+    @Override
+    public void setGenerateDefaultValues(String docTypeName) {
+        super.setGenerateDefaultValues(docTypeName);
+
+        List<Note> notes = new ArrayList<Note>();
+
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+
+            if (notes.isEmpty()) {
+                Note newBONote = getNewBoNoteForAdding(KFSConstants.CreateAndUpdateNotePrefixes.ADD);
+                notes.add(newBONote);
+            }
+        }
+    }
+
+    @Override
+    public void setupNewFromExisting(MaintenanceDocument document, Map<String, String[]> parameters) {
+        super.setupNewFromExisting(document, parameters);
+
+        List<Note> notes = new ArrayList<Note>();
+
+        if (getBusinessObject().getObjectId() != null) {
+            NoteService noteService = KRADServiceLocator.getNoteService();
+            notes = noteService.getByRemoteObjectId(getBusinessObject().getObjectId());
+        }
+
+        setAssetCreateAndUpdateNote(notes, KFSConstants.CreateAndUpdateNotePrefixes.ADD);
+
+        document.setNotes(notes);
+    }
+
+    /**
+     * Checks whether the previous note was an "Add" with the same document number as this one
+     *
+     * @param prefix
+     *            String to determine if it is a note "Add" or a note "Change"
+     */
+    private void setAssetCreateAndUpdateNote(List<Note> notes, String prefix) {
+        boolean shouldAddNote = true;
+
+        if (prefix.equals(KFSConstants.CreateAndUpdateNotePrefixes.CHANGE)) {
```

Comment:
Any reason not to collapse conditions from 77 and 79 into one bi-conditional if clause? Same goes for combining 81 and 82, any reason why they can't be in one 'if'? This would drop your nested statements from 4 levels to just two; one of our code standards states nested blocks should be limited to 3 levels or less, when possible. Collapsing levels tends to make code more readable, thus more easily maintained.